mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/inject/ShowStepFragmentModule.java

```diff
@@ -32,95 +32,33 @@
 
 package org.sagebionetworks.research.mobile_ui.inject;
 
-import android.support.annotation.NonNull;
-import android.support.v4.app.Fragment;
-
-import dagger.multibindings.Multibinds;
-
-import org.sagebionetworks.research.mobile_ui.show_step.ShowStepFragment;
 import org.sagebionetworks.research.mobile_ui.show_step.view.ShowActiveUIStepFragment;
 import org.sagebionetworks.research.mobile_ui.show_step.view.ShowCompletionStepFragment;
 import org.sagebionetworks.research.mobile_ui.show_step.view.ShowCountdownStepFragment;
+import org.sagebionetworks.research.mobile_ui.show_step.view.ShowFormUIStepFragment;
 import org.sagebionetworks.research.mobile_ui.show_step.view.ShowStepFragmentBase;
 import org.sagebionetworks.research.mobile_ui.show_step.view.ShowUIStepFragment;
-import org.sagebionetworks.research.mobile_ui.show_step.view.ShowFormUIStepFragment;
-import org.sagebionetworks.research.presentation.inject.DrawableModule;
-import org.sagebionetworks.research.presentation.model.implementations.ActiveUIStepViewBase;
-import org.sagebionetworks.research.presentation.model.implementations.CompletionStepViewBase;
-import org.sagebionetworks.research.presentation.model.implementations.CountdownStepViewBase;
-import org.sagebionetworks.research.presentation.model.implementations.FormUIStepViewBase;
-import org.sagebionetworks.research.presentation.model.implementations.UIStepViewBase;
-import org.sagebionetworks.research.presentation.model.interfaces.StepView;
-
-import java.util.Map;
 
-import dagger.MapKey;
 import dagger.Module;
-import dagger.Provides;
-import dagger.multibindings.IntoMap;
+import dagger.android.ContributesAndroidInjector;
 
-// The ShowUIStepViewFragment needs the drawable module so it doesn't make sense to use one of these modules without
-// the other.
-@Module(includes = DrawableModule.class)
+@Module
 public abstract class ShowStepFragmentModule {
-    @MapKey
-    public @interface StepViewKey {
-        String value();
-    }
-
-    @Multibinds
-    abstract Map<String, ShowStepFragmentFactory> fragmentFactoryMap();
-
-    public interface ShowStepFragmentFactory {
-        @NonNull
-        ShowStepFragment create(@NonNull StepView stepView);
-    }
-
-    @Provides
-    public static ShowStepFragmentFactory provideShowStepFragmentFactory(
-            Map<String, ShowStepFragmentFactory> showStepFragmentFactoryMap) {
-        return (@NonNull StepView stepView) -> {
-            if (showStepFragmentFactoryMap.containsKey(stepView.getType())) {
-                return showStepFragmentFactoryMap.get(stepView.getType()).create(stepView);
-            }
-
-            // If we don't have a factory we default to the most general ShowStepFragment.
-            return ShowUIStepFragment.newInstance(stepView);
-        };
-    }
+    @ContributesAndroidInjector
```

Comment:
@ContributesAndroidInjector is infinitely better than the manual way I was doing it before.